pkg/k8shandler/fluentd.go

```diff
@@ -214,43 +220,90 @@ func newFluentdPodSpec(logging *logging.ClusterLogging, elasticsearchAppName str
 		},
 	}
 
-	fluentdContainer.Env = []v1.EnvVar{
-		{Name: "NODE_NAME", ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "spec.nodeName"}}},
-		{Name: "MERGE_JSON_LOG", Value: "false"},
-		{Name: "PRESERVE_JSON_LOG", Value: "true"},
-		{Name: "K8S_HOST_URL", Value: "https://kubernetes.default.svc"},
-		{Name: "ES_HOST", Value: elasticsearchAppName},
-		{Name: "ES_PORT", Value: "9200"},
-		{Name: "ES_CLIENT_CERT", Value: "/etc/fluent/keys/app-cert"},
-		{Name: "ES_CLIENT_KEY", Value: "/etc/fluent/keys/app-key"},
-		{Name: "ES_CA", Value: "/etc/fluent/keys/app-ca"},
-		{Name: "METRICS_CERT", Value: "/etc/fluent/metrics/tls.crt"},
-		{Name: "METRICS_KEY", Value: "/etc/fluent/metrics/tls.key"},
-		{Name: "OPS_HOST", Value: elasticsearchInfraName},
-		{Name: "OPS_PORT", Value: "9200"},
-		{Name: "OPS_CLIENT_CERT", Value: "/etc/fluent/keys/infra-cert"},
-		{Name: "OPS_CLIENT_KEY", Value: "/etc/fluent/keys/infra-key"},
-		{Name: "OPS_CA", Value: "/etc/fluent/keys/infra-ca"},
-		{Name: "BUFFER_QUEUE_LIMIT", Value: "32"},
-		{Name: "BUFFER_SIZE_LIMIT", Value: "8m"},
-		{Name: "FILE_BUFFER_LIMIT", Value: "256Mi"},
-		{Name: "FLUENTD_CPU_LIMIT", ValueFrom: &v1.EnvVarSource{ResourceFieldRef: &v1.ResourceFieldSelector{ContainerName: "fluentd", Resource: "limits.cpu"}}},
-		{Name: "FLUENTD_MEMORY_LIMIT", ValueFrom: &v1.EnvVarSource{ResourceFieldRef: &v1.ResourceFieldSelector{ContainerName: "fluentd", Resource: "limits.memory"}}},
-		{Name: "NODE_IPV4", ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "status.hostIP"}}},
-		{Name: "CDM_KEEP_EMPTY_FIELDS", Value: "message"}, // by default, keep empty messages
+	if (proxyConfig != nil) {
```

Comment:
Please try to figure out a way to define the `fluentdContainer.Env` once, then dynamically append the proxy specific fields if `proxyConfig != nil`

Reply:
Well, I did so using append like this [1] first...  By calling this func, the proxy env vars are added to the array of v1.EnvVar (I verified that by dumping the array), but the proxy env vars were not passed to the fluentd pod.  (that tricked me to consider exporting them in the fluentd run.sh...)  Do you think there's some method to ensure setting the env vars dynamically set?

[1]
```
func updateEnvVar(env *[]v1.EnvVar, m v1.EnvVar) {
    exists := false
    for _, elem := range *env {
        if elem.Name == m.Name {
            exists = true
            if elem.Value != m.Value {
                elem.Value = m.Value
                fmt.Printf("fluentd: set env(%s, %s) - replace\n", m.Name, m.Value)
            }
        }
    }
    if !exists {
        *env = append(*env, m)
        fmt.Printf("fluentd: set env(%s, %s) - new\n", m.Name, m.Value)
    }
}

```

Reply:
I don't think you'd need to use the function you could do something like:
```
fluentdEnv := []v1.EnvVar{
  // normal env var setting
}

if proxyConfig != nil {
  proxyEnv := []v1.EnvVar{
    {Name: "HTTPS_PROXY" ...
   ... 
  }

  fluentdEnv = append(fluentdEnv, proxyEnv...)
  // note the "..." after proxyEnv, this is required because you're appending a list to a list
}

fluentdContainer.Env = fluentdEnv
```

Reply:
Yes - something like that should work - not sure what's wrong - I don't know enough about go or the operator sdk to spot the error

Reply:
You can use the above pattern for the volumes and volume mounts as well

Reply:
Thanks!  Let me apply your patch...

Reply:
also - go style is no parentheses - so use `if proxyConfig != nil {`